.github/workflows/run-pytest.yml

```diff
@@ -30,22 +30,23 @@ jobs:
           - os: 'macos-latest'
             python: '3.12'
     steps:
-    - uses: actions/checkout@v4
+    - name: Checkout
+      uses: actions/checkout@a5ac7e51b41094c92402da3b24376905380afc29 # v4.1.6
       with:
         submodules: "recursive"
     - name: Set up Python ${{ matrix.python }}
-      uses: actions/setup-python@v5
+      uses: actions/setup-python@39cd14951b08e74b54015e9e001cdefcf80e669f # v5.1.1
       with:
         python-version: ${{ matrix.python }}
-    - name: Update pip/build packages
-      run: |
-        pip install setuptools --upgrade
-    - name: Install dependencies
-      run: |
-        pip install -r .github/workflows/requirements-test.txt
-    - name: Install the package
+    - name: Set up Python environment
+      uses: glotzerlab/workflows/setup-uv@3573ddaebf3290e00bfb150b82a46412192a61d3
+      with:
+        only-binary: ":none:"
+        lockfile: ".github/requirements${{ matrix.python }}.txt"
+    - name: Install package
       run: |
-        pip install -e .
+        python --version
+        uv pip install . --no-build-isolation --no-deps --system -v
```

Comment:
where is pytest installed?

Reply:
Environment is set up in the setup-UV action, with a `UV pip sync` under the hood. It's a break from our previous workflow but it's a net positive IMO